File: Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/util/wrappers/OptionsMenuViewInteraction.java
package org.catrobat.paintroid.test.espresso.util.wrappers;

import android.view.View;

import org.hamcrest.CoreMatchers;

import androidx.annotation.StringRes;
import androidx.appcompat.widget.MenuPopupWindow;
import androidx.test.espresso.ViewInteraction;

import static org.catrobat.paintroid.test.espresso.util.UiMatcher.withAdaptedData;
import static org.hamcrest.Matchers.not;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;

public final class OptionsMenuViewInteraction {
	static ViewInteraction optionsMenu;

	private OptionsMenuViewInteraction() {
		optionsMenu = onView(CoreMatchers.<View>instanceOf(MenuPopupWindow.MenuDropDownListView.class));
	}

	public static OptionsMenuViewInteraction onOptionsMenu() {
		return new OptionsMenuViewInteraction();
	}

	public OptionsMenuViewInteraction checkItemExists(@StringRes int item) {
		optionsMenu.check(matches(withAdaptedData(item)));

		return this;
	}

	public OptionsMenuViewInteraction checkItemDoesNotExist(@StringRes int item) {
		optionsMenu.check(matches(not(withAdaptedData(item))));
		return this;
	}
}

File: Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/util/wrappers/OptionsMenuViewInteraction.kt
package org.catrobat.paintroid.test.espresso.util.wrappers

import androidx.annotation.StringRes
import androidx.appcompat.widget.MenuPopupWindow.MenuDropDownListView
import androidx.test.espresso.Espresso
import androidx.test.espresso.ViewInteraction
import androidx.test.espresso.assertion.ViewAssertions
import org.catrobat.paintroid.test.espresso.util.UiMatcher
import org.hamcrest.CoreMatchers
import org.hamcrest.Matchers

class OptionsMenuViewInteraction private constructor() {
    init {
        optionsMenu = Espresso.onView(
            CoreMatchers.instanceOf(
                MenuDropDownListView::class.java
            )
        )
    }

    fun checkItemExists(@StringRes item: Int): OptionsMenuViewInteraction {
        optionsMenu.check(ViewAssertions.matches(UiMatcher.withAdaptedData(item)))
        return this
    }

    fun checkItemDoesNotExist(@StringRes item: Int): OptionsMenuViewInteraction {
        optionsMenu.check(ViewAssertions.matches(Matchers.not(UiMatcher.withAdaptedData(item))))
        return this
    }

    companion object {
        lateinit var optionsMenu: ViewInteraction
        fun onOptionsMenu(): OptionsMenuViewInteraction = OptionsMenuViewInteraction()
    }
}
